reactor-core/src/main/java/reactor/core/publisher/Operators.java

```diff
@@ -376,6 +377,109 @@ public static RuntimeException onRejectedExecution(Throwable original, Context c
 		return onRejectedExecution(original, null, null, null, context);
 	}
 
+	static final OnNextFailureStrategy onNextErrorStrategy(Context context) {
+		OnNextFailureStrategy strategy = null;
+
+		BiFunction<? super Throwable, Object, ? extends Throwable> fn = context.getOrDefault(
+				OnNextFailureStrategy.KEY_ON_NEXT_ERROR_STRATEGY, null);
+		if (fn instanceof OnNextFailureStrategy) {
+			strategy = (OnNextFailureStrategy) fn;
+		} else if (fn != null) {
+			strategy = new OnNextFailureStrategy.LambdaOnNextErrorStrategy(fn);
+		}
+
+		if (strategy == null) strategy = Hooks.onNextErrorHook;
+		if (strategy == null) strategy = OnNextFailureStrategy.STOP;
+		return strategy;
+	}
+
+	public static final BiFunction<? super Throwable, Object, ? extends Throwable> onNextErrorFunction(Context context) {
```

Comment:
This still need tying in?

Reply:
yes, we don't want to expose a new interface for now, but the interface allows for smarter checks for internal implementations when it is used as a `BiPredicate`.

Reply:
Also, looks like this would be the way to support context in the future too..